src/lumigo_tracer/user_utils.py

```diff
@@ -1,12 +1,41 @@
+import json
+from typing import Dict, Optional
+
 from lumigo_tracer.spans_container import SpansContainer
 from lumigo_tracer.lumigo_utils import Configuration, warn_client
 
 LUMIGO_REPORT_ERROR_STRING = "[LUMIGO_LOG]"
 MAX_TAGS = 50
+MAX_ELEMENTS_IN_EXTRA = 10
 MAX_TAG_KEY_LEN = MAX_TAG_VALUE_LEN = 50
 ADD_TAG_ERROR_MSG_PREFIX = "Skipping add_execution_tag: Unable to add tag"
 
 
+def info(msg: str, error_type: str = "ProgrammaticInfo", extra: Dict[str, str] = None):
+    log(20, msg, error_type, extra)
+
+
+def warn(msg: str, error_type: str = "ProgrammaticWarn", extra: Dict[str, str] = None):
+    log(30, msg, error_type, extra)
+
+
+def error(msg: str, error_type: str = "ProgrammaticError", extra: Dict[str, str] = None):
+    log(40, msg, error_type, extra)
+
+
+def log(level: int, msg: str, error_type: str, extra: Optional[Dict[str, str]]):
+    tags_len = SpansContainer.get_span().get_tags_len()
+    extra_filtered = list(
+        filter(
+            lambda element: validate_tag(element[0], str(element[1]), tags_len, True),
```

Comment:
```suggestion
            lambda element: validate_tag(element[0],element[1], tags_len, True),
```
And change `validate_tag` to accept non-string inputs